src/hiveboard_bridge/src/MessageHandler.cpp

```diff
@@ -0,0 +1,46 @@
+#include "hiveboard_bridge/MessageHandler.h"
+
+MessageHandler::MessageHandler() {}
+
+MessageHandler::~MessageHandler() {}
+
+bool MessageHandler::handleMessage(MessageDTO message) {
+    std::variant<std::monostate, RequestDTO> request = message.getMessage();
+
+    if (std::holds_alternative<RequestDTO>(request)) {
+        std::variant<std::monostate, FunctionCallRequestDTO> functionCallRequest =
+            std::get<RequestDTO>(request).getRequest();
+
+        if (std::holds_alternative<FunctionCallRequestDTO>(functionCallRequest)) {
+            FunctionCallRequestDTO function = std::get<FunctionCallRequestDTO>(functionCallRequest);
+            std::string functionName = function.getFunctionName();
+            CallbackArgs functionArgs = function.getArguments();
+            uint16_t argsLength = function.getArgumentsLength();
```

Comment:
There has been a fair bit of refactoring so it is possible that I forgot to remove some stuff.

However, in this case, I figured that the contents of the function calls will be _well-known_, since this is user code. It is the user's job to implement a callback with arguments that match the ones that were declared in the HiveMind/Buzz, isn't it? In that case, the callbacks do not need to know the number of args, it already knows that `arg[0]` means X and that `arg[1]` means Y (see the example callbacks declared in the unit tests for MessageHandler).

I could still pass along the `argsLength` if we want to provide a way for the users to do some checks in their callbacks, but I just wanted to expose the fact that the argsLength should not have any "applicative" usage; it should merely be used for safety measures.

What do you think?

Reply:
Yes, but it would be easier to check (i.e. faster) then length of the arguments than if there is enough arguments by checking one by one. It makes it easier for the user to debug since it's possible he could have multiple buzz script version by accident, causing issues with the number of arguments to not match between versions.

It would make it explicit to not overflow the buffer too.

But I see what you mean with the applicative usage, though it would be more safe to pass it. Making the intent explicit and avoiding out of bounds reads.